libbeat/processors/actions/lowercase_fields.go

```diff
@@ -0,0 +1,72 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package actions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	"github.com/elastic/beats/v7/libbeat/processors/checks"
+	conf "github.com/elastic/elastic-agent-libs/config"
+)
+
+func init() {
+	processors.RegisterPlugin(
+		"lowercase_keys",
+		checks.ConfigChecked(
+			NewLowerCaseKeyProcessor,
+			checks.RequireFields("fields"),
+			checks.AllowedFields("fields", "when", "ignore_missing", "fail_on_error"),
+		),
+	)
+}
+
+// NewLowerCaseKeyProcessor converts event keys matching the provided fields to lowercase
+func NewLowerCaseKeyProcessor(c *conf.C) (processors.Processor, error) {
+	return NewChangeFieldProcessor(c, "lower", lowerCaseKey)
```

Comment:
This name passed here (`"lower"`) must match the actual processor name when the plugin is registered, it's required for logging.